PathKit/Sources/LineSegment.swift

```diff
@@ -27,13 +27,64 @@ public struct LineSegment {
         return lerp(start, end, at: t)
     }
 
-    /// Creates a line segment begins at the point `start` and ends the point `end`.
+    /// Creates a line segment beginning at `start` and ending at `end`.
     ///
     /// - Parameters:
     ///   - start: The starting point of the line segment.
     ///   - end: The end point of the line segment.
+    /// - Precondition: The `start` point and `end` point of the line segment must be different.
     public init(start: Point, end: Point) {
+        precondition(start != end)
         self.start = start
-        self.end = start
+        self.end = end
+    }
+
+
+    public func offset(by vector: Vector) -> LineSegment {
+        return LineSegment(start: start + vector, end: end + vector)
+    }
+
+    /// Computes the intersection of two line segments.
+    /// Source: https://stackoverflow.com/questions/563198/how-do-you-detect-where-two-line-segments-intersect
+    public static func intersection(_ lhs: LineSegment, _ rhs: LineSegment, accuracy: Double) -> Intersection {
+        let p = lhs.start
+        let q = rhs.start
+        let qp = Vector(head: q, tail: p)
+        let r = Vector(head: lhs.end, tail: lhs.start)
+        let s = Vector(head: rhs.end, tail: rhs.start)
+        let parallel = Vector.areParallel(r, s, accuracy: accuracy)
+        let collinear = Vector.areParallel(qp, r, accuracy: accuracy)
+        let zeroToOne = 0.0...1.0
+
+        if parallel && collinear {
+            // lines are collinear
+            let unitVectorR = r.unitVector
+            let t0 = qp • unitVectorR
+            let t1 = t0 + s • unitVectorR
+            let range = (s • r < 0) ? t1...t0 : t0...t1
+
+            if range.overlaps(zeroToOne) {
+                // line segments overlap
+                return .infinite
+            } else {
+                // line segments are disjoint
+                return .empty
+            }
+        } else if parallel {
+            // lines are parallel and non-intersecting
+            return .empty
+        } else {
+            let rXs = r ✕ s
+            let t = qp ✕ s / rXs
+            let u = qp ✕ r / rXs
+
+            if zeroToOne.contains(t) && zeroToOne.contains(u) {
```

Comment:
I think the accuracy should be taken into account here?

Reply:
Good catch. Will fix